### PR TITLE
(closes #1383) Standardise and cleanup unneeded dag_name methods

### LINKED ISSUE #1383: Standardise Node output name between methods to remove boilerplate code
Some nodes (specifically Directives and some PSyKAl concepts) have slightly different names for the: name class, the str/view representation and the dag_name. e.g. the OMPParallelDirective class has:
- class name: OMPParallelDirective
- str/view representation: Directive[OMPParallel]
- dag_name: OMP_parallel_

These small changes create a lot of boilerplate code in the classes to explain how to produce each string. I think it will be better to standardise the str/view node name and the dag name to use the class name so that a generic implementation of these in the base node class would be enough in most cases (the is still the parameters inside [] but this depend on the class so are fine).

### PR DESCRIPTION
**Pull request by sergisiso**

### CODE REVIEW

**Comment by sergisiso** [comment]
@arporter @rupertford This is ready to review. It's the second PR to close #1383 , similar to the previous one I made the generic dag_name work as expected by most classes and then do a `grep -r "def dag_name" -A 10` and clean up all the methods that didn't do anything additional.

File: src/psyclone/domain/gocean/nodes/gocean_extract_node.py [hunk 1]
```diff
@@ -53,36 +54,7 @@ class GOceanExtractNode(ExtractNode):
     It adds a function 'generate_driver' which creates a GOcean-specific
     stand-alone driver program that can read the created output files.
```
Comment (arporter) [inline src/psyclone/domain/gocean/nodes/gocean_extract_node.py:55]: This begs the question: do we need to subclass ExtractNode if we're not adding any functionality to it? I would suggest not but open the floor to @hiker and @rupertford for discussion.

File: src/psyclone/psyir/nodes/node.py [hunk 2]
```diff
@@ -623,8 +623,23 @@ def dag_gen(self, graph):
 
     @property
     def dag_name(self):
-        '''Return the base dag name for this node.'''
-        return "node_" + str(self.abs_position)
+        '''Return the dag name for this node. This includes the name of the
+        class and a index of its relative position to the parent Routine. If
```
Comment (arporter) [inline src/psyclone/psyir/nodes/node.py:627]: s/a index/the index/

File: src/psyclone/tests/psyir/nodes/node_test.py [hunk 3]
```diff
@@ -506,7 +506,12 @@ def test_dag_names():
     psy = PSyFactory("dynamo0.3", distributed_memory=True).create(invoke_info)
     invoke = psy.invokes.invoke_list[0]
     schedule = invoke.schedule
-    assert super(Schedule, schedule).dag_name == "node_1"
+
+    # Classes without the dag_name specialised should show the name of the
+    # class and the relative position to the ancestor routine
+    assert schedule.children[4].start_expr.dag_name == "Literal_6"
+
+    # Some classes have their won specialisation of the dag_name
```
Comment (arporter) [inline src/psyclone/tests/psyir/nodes/node_test.py:514]: s/won/own/

**Review by arporter — CHANGES_REQUESTED** [review]
Looks good Sergi.
All tests and examples OK with compilation.
All coverage is OK.
All docs build without new warnings.
Only major thing is whether we should remove `GOceanExtractNode` altogether as it doesn't implement any new functionality.
All changes are pycodestyle/pylint clean.

File: src/psyclone/tests/psyir/nodes/node_test.py (hunk 3 above)
Comment (sergisiso) [reply src/psyclone/tests/psyir/nodes/node_test.py:514]: Done

File: src/psyclone/psyir/nodes/node.py (hunk 2 above)
Comment (sergisiso) [reply src/psyclone/psyir/nodes/node.py:627]: Done

**Comment by sergisiso** [comment]
@arporter This is ready for another look.

**Review by arporter — CHANGES_REQUESTED** [review]
All requested changes have been made.
All tests and examples are OK with compilation.
All docs build without new warnings.
I'd taken the liberty of bringing this up-to-date with master but then I discovered that there are still two TODOs referencing 1383 in `directive_test.py`. Please could you investigate.

**Comment by sergisiso** [comment]
@arporter Aha, this are meant to be referencing #1388, updated now. I also pinned jsonschema to fix CI issues with older python. This is ready for another look.

**Pull request merged**